e2e/jest.config.js

```diff
@@ -1,7 +1,11 @@
+const path = require("path")
+
 module.exports = {
   preset: "jest-playwright-preset",
-  setupFilesAfterEnv: ["expect-playwright"],
-  testEnvironment: "./config/environment.js",
-  testRunner: "jest-circus/runner",
+  reporters: ["@testim/root-cause-jest/lib/reporter/default"],
+  testEnvironment: "@testim/root-cause-jest/lib/RootCauseJestEnv",
+  testEnvironmentOptions: {
+    actualEnvironment: path.resolve("./config/environment.js"),
```

Comment:
Hmm, this feels odd.  What if the root cause environment was an extendable class similar to how jest-playwright works?  That way you can use the default test environment or extend the class to provide your own customizations to the root cause test environment.

Reply:
You can still add your own customisations
The relevant coed is:
https://github.com/testimio/root-cause/blob/main/packages/root-cause-jest/lib/RootCauseJestEnv.ts#L108

I will try to explain why it's like that:

We (Root Cause) need to run code on environment level, and we want to make the integration as simple as possible.

But in many cases:
- The user will already have env from jest-playwright/puppeteer 
- The user will have custom env that extending , them

If we would expose a class that the user will need to extend from, It's less optimal because:
- We will need to provide one env that extends from puppeteer env, one from playwright, and one with nothing
- The user will have to change his env code as part of the integration
- Another path we've explored is mixing class https://github.com/testimio/root-cause/blob/main/packages/root-cause-jest/lib/circusRelated.ts#L35-L74

So what we've end up with is, we provide single env that can work for all cases, by loading the "actual env" inside it.
[It tries to "Guess" what env you want to use](https://github.com/testimio/root-cause/blob/main/packages/root-cause-jest/lib/RootCauseJestEnv.ts#L108) In a way that will make it to "just work" if you use jest-playwright/jest-puppeteer.

I this project, you have a custom env that we can't "infer" so we need the be told what env to load

I hope it went out clear :)